--- origsrc/telepathy-sharp/configure.ac	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/configure.ac	2007-06-08 14:09:08.258392600 -0500
@@ -26,7 +26,7 @@
 
 AC_SUBST(GACUTIL)
 
-GACUTIL_FLAGS='/gacdir $(DESTDIR)$(prefix)'
+GACUTIL_FLAGS='/gacdir $(DESTDIR)$(prefix)/lib'
 AC_SUBST(GACUTIL_FLAGS)
 
 
--- origsrc/telepathy-sharp/telepathy/Makefile.am	2007-06-08 13:41:47.000000000 -0500
+++ src/telepathy-sharp/telepathy/Makefile.am	2007-06-08 14:09:42.415079800 -0500
@@ -2,9 +2,9 @@
 
 ASSEMBLY_NAME = INdT.Telepathy
 ASSEMBLY = $(ASSEMBLY_NAME).dll
-INdT_Telepathydir = $(pkglibdir)
+INdT_Telepathydir = $(libdir)/mono/$(PACKAGE)
 INdT_Telepathy_SCRIPTS = $(ASSEMBLY)
-SNK=../telepathy-sharp.snk
+SNK=$(top_srcdir)/telepathy-sharp.snk
 
 TARGET = $(ASSEMBLY)
 
@@ -44,7 +44,7 @@
 	$(NDESK_DBUS_LIBS)
 
 $(ASSEMBLY): $(SOURCES) 
-	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $(SOURCES) $(PKG_REFERENCES) -keyfile:$(SNK)
+	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $^ $(PKG_REFERENCES) -keyfile:$(SNK)
 
 all: $(ASSEMBLY)
 
--- origsrc/telepathy-sharp/telepathy-sharp.pc.in	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/telepathy-sharp.pc.in	2007-06-08 14:12:37.432945000 -0500
@@ -6,5 +6,5 @@
 Description: Telepathy# - The INdT Telepathy .NET library
 Version: @VERSION@
 Cflags: 
-Libs: -r:${libdir}/telepathy-sharp/INdT.Telepathy.dll
+Libs: -r:${libdir}/mono/telepathy-sharp/INdT.Telepathy.dll
 Requires: ndesk-dbus-1.0
--- origsrc/telepathy-sharp/test/Makefile.am	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/test/Makefile.am	2007-06-08 14:03:48.441799000 -0500
@@ -12,7 +12,7 @@
 all-local: $(TARGET)
 
 $(TARGET): $(test_sources)
-	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $(test_sources) \
+	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $^ \
 	-target:exe -out:"$@"
 
 CLEANFILES = *.exe*
