diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/telepathy-sharp/configure.ac src/telepathy-sharp/configure.ac
--- origsrc/telepathy-sharp/configure.ac	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/configure.ac	2007-06-08 14:09:08.258392600 -0500
@@ -26,7 +26,7 @@
 
 AC_SUBST(GACUTIL)
 
-GACUTIL_FLAGS='/gacdir $(DESTDIR)$(prefix)'
+GACUTIL_FLAGS='/gacdir $(DESTDIR)$(prefix)/lib'
 AC_SUBST(GACUTIL_FLAGS)
 
 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/telepathy-sharp/telepathy/Makefile.am src/telepathy-sharp/telepathy/Makefile.am
--- origsrc/telepathy-sharp/telepathy/Makefile.am	2007-06-08 13:41:47.000000000 -0500
+++ src/telepathy-sharp/telepathy/Makefile.am	2007-06-08 14:09:42.415079800 -0500
@@ -2,9 +2,9 @@
 
 ASSEMBLY_NAME = INdT.Telepathy
 ASSEMBLY = $(ASSEMBLY_NAME).dll
-INdT_Telepathydir = $(pkglibdir)
+INdT_Telepathydir = $(libdir)/mono/$(PACKAGE)
 INdT_Telepathy_SCRIPTS = $(ASSEMBLY)
-SNK=../telepathy-sharp.snk
+SNK=$(top_srcdir)/telepathy-sharp.snk
 
 TARGET = $(ASSEMBLY)
 
@@ -44,7 +44,7 @@
 	$(NDESK_DBUS_LIBS)
 
 $(ASSEMBLY): $(SOURCES) 
-	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $(SOURCES) $(PKG_REFERENCES) -keyfile:$(SNK)
+	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $^ $(PKG_REFERENCES) -keyfile:$(SNK)
 
 all: $(ASSEMBLY)
 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/telepathy-sharp/telepathy-sharp.pc.in src/telepathy-sharp/telepathy-sharp.pc.in
--- origsrc/telepathy-sharp/telepathy-sharp.pc.in	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/telepathy-sharp.pc.in	2007-06-08 14:12:37.432945000 -0500
@@ -6,5 +6,5 @@
 Description: Telepathy# - The INdT Telepathy .NET library
 Version: @VERSION@
 Cflags: 
-Libs: -r:${libdir}/telepathy-sharp/INdT.Telepathy.dll
+Libs: -r:${libdir}/mono/telepathy-sharp/INdT.Telepathy.dll
 Requires: ndesk-dbus-1.0
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/telepathy-sharp/test/Makefile.am src/telepathy-sharp/test/Makefile.am
--- origsrc/telepathy-sharp/test/Makefile.am	2007-06-08 13:41:48.000000000 -0500
+++ src/telepathy-sharp/test/Makefile.am	2007-06-08 14:03:48.441799000 -0500
@@ -12,7 +12,7 @@
 all-local: $(TARGET)
 
 $(TARGET): $(test_sources)
-	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $(test_sources) \
+	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $^ \
 	-target:exe -out:"$@"
 
 CLEANFILES = *.exe*
